sway-core/tests/ir_to_asm/let_reassign_while_loop.asm

```diff
@@ -12,10 +12,12 @@ lw   $r0 data_0               ; literal instantiation
 sw   $r2 $r0 i0               ; store value
 addi $r0 $r2 i0               ; get_ptr
 lw   $r0 $r2 i0               ; load value
-jnei $r0 $one i21
+jnzi $r0 i15
+ji   i23
```

Comment:
So we're unfortunately adding an extra jump here with the benefit of 18bit addressing.  Not the end of the world, but we could one day be smarter and use `JNEI` when we know the offset is < 256K.  Can't really be done in IR, would have to be a peephole ASM optimisation.

Just a thought.  There are approximately a zillion of these things which we could add in the future and I'm guessing never will actually get around to doing.

Reply:
Yes this is a bit unfortunate. One alternative is to add another instruction that flips the condition before doing any jumps. That would get us back the same pattern of jumps as before but at the cost of an additional comparator. Or, as you said, use `jnei` if the "size" of the label fits in 12 bits.

I created a low priority issue here https://github.com/FuelLabs/sway/issues/1439